### PR TITLE
Personal/namalu/add connection and client protocols

### PR DESCRIPTION
**Pull request by namalu**
Added client and connection protocols and stubbed out several authentication methods.

### CODE REVIEW

File: MHVLib/Clients/MHVPersonClientProtocol.h [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#import "MHVClientProtocol.h"
+
+@class MHVApplicationSettings, MHVPersonInfo, MHVHealthRecordInfo;
+
```
Comment (miburf) [inline MHVLib/Clients/MHVPersonClientProtocol.h:22]: Can wrap the protocol with NS_ASSUME_NONNULL (same for all the protocols)

File: MHVLib/Connection/MHVConnection.m [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+@end
+
+@implementation MHVConnection
+
+- (instancetype)initWithConfiguration:(MHVConfiguration *)configuration
+                     credentialClient:(id<MHVSessionCredentialClientProtocol>)credentialClient
+                          httpService:(id<MHVHttpServiceProtocol>)httpService
+{
+    self = [super init];
```
Comment (miburf) [inline MHVLib/Connection/MHVConnection.m:35]: Should assert the parameters to be safe

File: MHVLib/Connection/MHVConnectionFactoryInternal.m [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    if (self)
+    {
+        _connectionQueue = dispatch_queue_create("MHVConnectionFactoryInternal.connectionQueue", DISPATCH_QUEUE_SERIAL);
+        _completions = [NSMutableArray new];
+    }
+    
+    return self;
+}
+
+- (void)getOrCreatSodaConnectionWithConfiguration:(MHVConfiguration *_Nonnull)configuration
```
Comment (miburf) [inline MHVLib/Connection/MHVConnectionFactoryInternal.m]: nit, Create

File: MHVLib/Connection/MHVSessionCredential.h [hunk 4]
```diff
@@ -0,0 +1,22 @@
+//
+//  MHVSessionCredential.h
+//  MHVLib
+//
+//  Created by Nathan Malubay on 5/11/17.
+//  Copyright © 2017 Microsoft Corporation. All rights reserved.
+//
+
```
Comment (miburf) [inline MHVLib/Connection/MHVSessionCredential.h]: Needs Microsoft license.  Several more files need it too.

File: MHVLib/Connection/MHVSessionCredential.m [hunk 5]
```diff
@@ -0,0 +1,26 @@
+//
+//  MHVSessionCredential.m
+//  MHVLib
+//
+//  Created by Nathan Malubay on 5/11/17.
+//  Copyright © 2017 Microsoft Corporation. All rights reserved.
```
Comment (miburf) [inline MHVLib/Connection/MHVSessionCredential.m]: License

File: MHVLib/Connection/MHVSessionCredentialClient.m [hunk 6]
```diff
@@ -0,0 +1,101 @@
+//
+//  MHVSessionCredentialClient.m
+//  MHVLib
+//
+//  Created by Nathan Malubay on 5/11/17.
```
Comment (miburf) [inline MHVLib/Connection/MHVSessionCredentialClient.m]: License.  I search for my name to find them :)

File: MHVLib/Connection/MHVSodaConnectionProtocol.h [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+#import <Foundation/Foundation.h>
+#import "MHVConnectionProtocol.h"
+
+/**
+ A client connection to HealthVault
+ */
+@protocol MHVSodaConnectionProtocol <MHVConnectionProtocol>
+
+/**
+ Prompt the user to authorize additional records for use in this app.
```
Comment (miburf) [inline MHVLib/Connection/MHVSodaConnectionProtocol.h:28]: Will this show a webview?  Should have an optional view controller as a parameter if so

File: MHVLib/Methods/MHVMethod.h [hunk 8]
```diff
@@ -0,0 +1,305 @@
...
+ @return An instance of MHVMethod.
+ */
++ (MHVMethod *)subscribeToEvent;
+
+/**
+ Remove a subscription.
+ 
+ @return An instance of MHVMethod.
+ */
++ (MHVMethod *)unsubscribeToEvent;
```
Comment (miburf) [inline MHVLib/Methods/MHVMethod.h:299]: unsubscribeFromEvent ?

File: MHVLib/Methods/MHVMethod.m [hunk 9]
```diff
@@ -0,0 +1,477 @@
+//
+//  MHVMethod.m
+//  MHVLib
+//
+//  Created by Nathan Malubay on 5/11/17.
```
Comment (miburf) [inline MHVLib/Methods/MHVMethod.m]: License

**Review by miburf — APPROVED** [review]

File: MHVLib/Connection/MHVSodaConnectionProtocol.h (hunk 7 above)
Comment (namalu) [reply MHVLib/Connection/MHVSodaConnectionProtocol.h:28]: No. The MHVConnectionProtocol handles the web view.

File: MHVLib/Methods/MHVMethod.h (hunk 8 above)
Comment (namalu) [reply MHVLib/Methods/MHVMethod.h:299]: Nope. It's mapped to documentation on developer.healthvault.com

**Pull request merged**